flatten_json.py

```diff
@@ -335,7 +335,7 @@ def _unflatten(dic, keys, value):
     list_keys = sorted(flat_dict.keys())
     for i, item in enumerate(list_keys):
         if i != len(list_keys) - 1:
-            if not list_keys[i + 1].startswith(list_keys[i]):
+            if not list_keys[i + 1].startswith(list_keys[i]+'.'):
```

Comment:
This only accounts when separator=`.`. I think it should be `if not list_keys[i + 1].startswith(list_keys[i]+separator):`